src/services/codeFixProvider.ts

```diff
@@ -124,9 +125,15 @@ export function eachDiagnostic(context: CodeFixAllContext, errorCodes: readonly
 }
 
 function getDiagnostics({ program, sourceFile, cancellationToken }: CodeFixContextBase) {
-    return [
+    const diagnostics = [
         ...program.getSemanticDiagnostics(sourceFile, cancellationToken),
         ...program.getSyntacticDiagnostics(sourceFile, cancellationToken),
         ...computeSuggestionDiagnostics(sourceFile, program, cancellationToken),
     ];
+    if (getEmitDeclarations(program.getCompilerOptions())) {

```

Comment:
If currently we are supporting providing code fix for errors only when `--isolatedDeclarations` is turned on may be should get declaration diagnostics only if its on to avoid perf hit

Reply:
I had them change to this in https://github.com/microsoft/TypeScript/pull/58260#discussion_r1585002609, though; in the editor, we'll have already called this for diags, and it's cached.

Reply:
We can flip it back to just isolatedDeclarations if you think it'll be a problem, but it seemed less error prone.

Reply:
Quoting Jake from the [other comment](https://github.com/microsoft/TypeScript/pull/58260/#discussion_r1585002609) 

> Actually, now that I think about it, this should probably just be getEmitDeclarations(program.getCompilerOptions()). It's possible that we could have quick fixes for declaration errors that are not in isolated declarations. And this call is cached so should be available for free anyway in codebases with declaration mode enabled.

Would you agree?

Reply:
Oh, my comment was slightly late 😅

Reply:
The cache does help a little but not whole lot since every edit will have to throw out the errors and recalculate them

Reply:
But, we call this function already and offer them up as diagnostics in the editor; won't the cost already have been paid?

Reply:
Yes i forgot about the part where LS.getSemanticDiagnostics gets declaration diagnostics as well.

Reply:
Yeah, for posterity:

```
    function getSemanticDiagnostics(fileName: string): Diagnostic[] {
        synchronizeHostData();

        const targetSourceFile = getValidSourceFile(fileName);

        // Only perform the action per file regardless of '-out' flag as LanguageServiceHost is expected to call this function per file.
        // Therefore only get diagnostics for given file.

        const semanticDiagnostics = program.getSemanticDiagnostics(targetSourceFile, cancellationToken);
        if (!getEmitDeclarations(program.getCompilerOptions())) {
            return semanticDiagnostics.slice();
        }

        // If '-d' is enabled, check for emitter error. One example of emitter error is export class implements non-export interface
        const declarationDiagnostics = program.getDeclarationDiagnostics(targetSourceFile, cancellationToken);
        return [...semanticDiagnostics, ...declarationDiagnostics];
    }

```